### DISCUSSION TITLE
Can the SD-2-1 model (768 resolution) be fine-tuned with full-parameter finetuning?

### DISCUSSION THREAD

**Discussion by krahets** [huggingface/diffusers#9468] [General]
I conducted fine-tuning experiments on two base models, SD-2-1-base (512x512 resolution) and SD-2-1 (768x768 resolution), using identical hyperparameters:

```python
"pretrained_model_name_or_path": "stabilityai/stable-diffusion-2-1",  # Use "stabilityai/stable-diffusion-2-1-base" for the 512x512 model
"max_train_steps": 20000,
"train_batch_size": 1,
"gradient_accumulation_steps": 8,
"learning_rate": 3e-05,
"lr_scheduler": "constant",
"lr_warmup_steps": 0,
"max_grad_norm": 1,
"prediction_type": "v_prediction",
"mixed_precision": "bf16",
"gradient_checkpointing": true,
"enable_xformers_memory_efficient_attention": true,
"height": 768,
"width": 512,
```

**Model architecture**:

An image-conditioned model with extended input channels, where the conditioning image is concatenated with the noise (similar to the approach in Zero123), and the CLIP text encoder is disabled.

**Dataset**:

15k frames for a single person.

**Results:**

**SD-2-1-base (512x512 resolution):** 

(Above: Ground Truth (GT), Below: Model Generation)

<img width="168" alt="SD-2-1-base result" src="https://github.com/user-attachments/assets/075b548a-6811-47d6-9e2c-bfc64e9c5ab5">

**SD-2-1 (768x768 resolution):**

(Above: Ground Truth (GT), Below: Model Generation)

<img width="168" alt="SD-2-1 result" src="https://github.com/user-attachments/assets/4b195c6c-5481-499c-b8c5-ddc778d92dd7">

It looks like the SD-2-1 (768x768 resolution) faced model collapse.

**Comment by asomoza** [comment]
You can train SD 2.1 with 768px images and should work, probably you need to adapt the params for it. 

One example of this is [SPRIGHT ](https://spright-t2i.github.io/) that was trained with 768px images with this [dataset](https://huggingface.co/datasets/SPRIGHT-T2I/spright).

cc: @sayakpaul if you have some more insights

**Reply by krahets** [reply]
@asomoza Thank you!

I was developing an image-conditioned model with extended input channels, where the conditioning image and camera pose are concatenated with the noise (similar to the approach in [Zero123](https://zero123.cs.columbia.edu/)), and the CLIP text encoder is disabled.

I wonder if it's feasible to fine-tune this channel-extended model from [SD 2.1 768px](https://huggingface.co/stabilityai/stable-diffusion-2).

**Reply by asomoza** [reply]
I don't see why not, I didn't test SD 2 or 2.1 that much but it should be finetuneable with more channels. The image resolution shouldn't be a problem with it also.

Still, SD 2 was not popular and used because it often produced bad results, watermarks and bad anatomy. Taking into account your demo images you seem to be training with humans which is the worse use case for SD 2 or 2.1

You're probably safer just using SD 1.5 for what you want.

**Reply by krahets** [reply]
@asomoza  I’ve noticed that the human priors in SD2/2.1 are poor, and they rarely generate high-quality full-body human images. I suspect this is due to the training dataset being filtered for NSFW content. What do you think?

<img width="408" alt="image" src="https://github.com/user-attachments/assets/2ee1cba6-fcd0-478c-ae20-ddffd25e9268">

Recently, I’ve found that SD3-medium can produce high-quality human images. Do you think it would be safe to perform full-parameter fine-tuning based on SD3-medium?

**Reply by asomoza** [reply]
yeah, it was because of the NSFW safety filter, sadly SD3 suffered with the same. You can generate good humans with standing poses with SD3 but if you ask it for something more elaborated it outputs really weird stuff. You can just search for "sd3 woman lying on the grass". I really like SD3 for the photorealism generations though and I use it a lot.

If you want to go with human anatomy and poses, you should use SDXL or SD 1.5.

**Reply by bghira** [reply]
bytedance released a finetuned 2.1 model you could try.

**Reply by krahets** [reply]
@bghira Thanks! Are you referring to this model [sd2.1-base-zsnr-laionaes5](https://huggingface.co/ByteDance/sd2.1-base-zsnr-laionaes5)?

**Reply by bghira** [reply]
yes, it has a superior noise schedule to the original and was reparameterised to v-prediction from epsilon, completely bypassing issues from SD 2.1-v

**Reply by krahets** [reply]
@bghira Cool! I tested  it with the prompts:

> A young girl smiling

<img width="509" alt="image" src="https://github.com/user-attachments/assets/69c1b5f5-375f-4fed-9759-b661aa7604fe">

> Woman lying on the grass

<img width="509" alt="image" src="https://github.com/user-attachments/assets/5bf22a48-ef3a-4ff5-abfd-2ea7ff4e04fb">

> Children playing in the park on a sunny day

<img width="509" alt="image" src="https://github.com/user-attachments/assets/0ff379f3-f510-4dfe-b3f5-2546208b57cb">

> Students in the classroom

<img width="509" alt="image" src="https://github.com/user-attachments/assets/30962bd9-5238-46c3-a6be-48e9168ca9ad">

It seems that the model also struggles with accurately representing human anatomy. 

Do you think it’s important to use a pretrained model with strong human priors? If so, could you recommend an option?

I don't know if my current thinking correct: 

- SD1.5/SD2.1: Poor human priors.
- SD3:  Struggling with accurately representing human anatomy.
- Flux: According to [your comments](https://github.com/black-forest-labs/flux/issues/9#issuecomment-2264151872), it's not tunable for full-parameter fine-tuning.

By the way, thank you for sharing **SimpleTuner**. It’s been a great help to the community!

**Reply by bghira** [reply]
haven't tried pixart or kolors?

**Reply by bghira** [reply]
also SD2.1 needs long prompts, and for realism, ideally a location on the planet and a year

`stunning photograph of a woman standing on the sidewalk in 1998 san francisco, kodachrome, vintage, influencer vibes`
